.github/workflows/tics.yaml

```diff
@@ -0,0 +1,61 @@
+name: TICS
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Verify snap builds successfully
+        id: build
+        uses: canonical/action-build@v1
+
+      - name: Upload the built snap
+        uses: actions/upload-artifact@v4
+        with:
+          name: snap_artifact
+          path: ${{ steps.build.outputs.snap }}
```

Comment:
The build already happens on check.yaml. Why are you proposing to build it again?

Reply:
we need the built artifact in order to run the test-all target, which is in turn needed to produce the coverage reports. That said, I don't think I like having this step here vs being a dependency within tox